org.eclipse.dawnsci.nexus/src/org/eclipse/dawnsci/nexus/NexusConstants.java

```diff
@@ -0,0 +1,147 @@
+	public static final String MIRROR = "NXmirror";
+	public static final String MODERATOR = "NXmoderator";
+	public static final String MONITOR = "NXmonitor";
+	public static final String MONOCHROMATOR = "NXmonochromator";
+	public static final String NOTE = "NXnote";
+	public static final String OBJECT = "NXobject";
+	public static final String ORIENTATION = "NXorientation";
+	public static final String PARAMETERS = "NXparameters";
+	public static final String PINHOLE = "NXpinhole";
+	public static final String POLARIZER = "NXpolarizer";
+	public static final String POSITIONER = "NXpositioner";
+	public static final String PROCESS = "NXprocess";
+	public static final String REFLECTIONS = "NXreflections";
+	public static final String ROOT = "NXroot";
+	public static final String SAMPLECOMPONENT = "NXsample_component";
+	public static final String SAMPLE = "NXsample";
+	public static final String SENSOR = "NXsensor";
+	public static final String SHAPE = "NXshape";
+	public static final String SLIT = "NXslit";
+	public static final String SOURCE = "NXsource";
+	public static final String SUBENTRY = "NXsubentry";
+	public static final String TRANSFORMATIONS = "NXtransformations";
+	public static final String TRANSLATION = "NXtranslation";
+	public static final String USER = "NXuser";
+	public static final String VELOCITYSELECTOR = "NXvelocity_selector";
+	public static final String XRAYLENS = "NXxraylens";
+
+	/**
+	 * NXdata default data name
+	 */
+	public static final String DATA_DATA = "data";
+	/**
+	 * NXdata dataset attribute
+	 */
+	public static final String DATA_NAME = "long_name";
+	/**
+	 * NXdata attribute that points to main dataset
+	 */
+	public static final String DATA_SIGNAL = "signal";
+	/**
+	 * NXdata attribute that maps dimensions of main dataset to axis datasets
+	 */
+	public static final String DATA_AXES = "axes";
+	/**
+	 * NXdata attribute that points to error datasets (may also be legacy if attached to main dataset)
+	 */
+	public static final String DATA_UNCERTAINTIES ="uncertainties";
+
+	/**
+	 * NXdata indicate no default for axis in axes attribute
+	 */
+	public static final String DATA_AXESEMPTY = ".";
+
+	/**
+	 * Suffix for name of error datasets
+	 */
+	public static final String DATA_ERRORS_SUFFIX = "_errors";
+	/**
+	 * Suffix for name of attributes that indicate dimension mappings from axis dataset to main dataset
+	 */
+	public static final String DATA_INDICES_SUFFIX = "_indices";
+	/**
+	 * Suffix for names of datasets
+	 */
+	public static final String DATA_UNCERTAINTY_SUFFIX ="_uncertainty";
+	/**
+	 * Suffix for names of datasets that contain set values (for axes)
+	 */
+	public static final String DATA_AXESSET_SUFFIX = "_set";
+
+	/**
+	 * @deprecated legacy attribute name of main dataset
+	 */
+	@Deprecated
+	public static final String DATA_ERRORS = "errors";
+	/**
+	 * @deprecated legacy attribute name
+	 */
+	@Deprecated
+	public static final String DATA_AXIS = "axis";
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1133)